### PR TITLE
[WIP] Stack onboarding

### PR DESCRIPTION
**Pull request by jvanderaa**
- Updates components to match new format
- Adds tests for the IOS parsing

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -85,15 +86,52 @@ During a successful onboarding process, a new device will be created in NetBox w
...
+individual devices. The primary (first) unit in the stack will:
+
+- Have the primary IP address associated (if configured, defaults to yes)
+- Will **not** have the `stack_separator` in the name of the device, by using colon (:) this would
+override any default application port with automation systems.
+
+All of the additional units will have the separator_indicator separating the name from the unit
+number. There **will not** be a primary IP address assigned to the unit. Assuming that there is a
+methodology to parse each of the model numbers and serial numbers from a command output, there will
+be added in about the device that is unique to the device of serial number, model number.
```
Comment (glennmatthews) [inline README.md:101]: Last sentence here is a bit confusing to me. Would the following be a correct restatement?

"If this plugin implements a methodology to parse per-unit information (such as serial number, model number, etc.) from a command output, then this information will be added to the unit."

File: README.md [hunk 2]
```diff
@@ -85,15 +86,52 @@ During a successful onboarding process, a new device will be created in NetBox w
...
+
+> The idea of the stacking is that there is a single management IP address for multiple switch
+> units. Thus there will not be hte primary IP and is tracked for physical inventory, but not for
```
Comment (glennmatthews) [inline README.md:104]: nit: hte/the, but this sentence is also confusing to me. Can you clarify it?

File: README.md [hunk 3]
```diff
@@ -85,15 +86,52 @@ During a successful onboarding process, a new device will be created in NetBox w
...
+| NetBox Device Name | Primary IP Address | Model Number     | Serial Number  | Notes                                            |
+| ------------------ | ------------------ | ---------------- | -------------- | ------------------------------------------------ |
+| nyc01-sw01         | 192.0.2.10/24      | Unique to Unit 1 | Unique to unit | This **should** be used for connecting to device |
+| nyc01-sw01:2       | -                  | Unique to Unit 2 | Unique to unit | No address will be available to connect to       |
+| nyc01-sw01:3       | -                  | Unique to Unit 3 | Unique to unit | No address will be available to connect to       |
+| nyc01-sw01:4       | -                  | Unique to Unit 4 | Unique to unit | No address will be available to connect to       |
+
+### Tested Stack Platforms
+
+These are the list of tested stacking platforms. This should help to provide confidence in what platforms can be
```
Comment (glennmatthews) [inline README.md:123]: Are you going to add a list here?

File: netbox_onboarding/netbox_keeper.py [hunk 4]
```diff
@@ -317,6 +319,7 @@ def ensure_device_instance(self, default_status=PLUGIN_SETTINGS["default_device_
         try:
             self.device, created = Device.objects.update_or_create(**lookup_args)
 
+
```
Comment (glennmatthews) [inline netbox_onboarding/netbox_keeper.py:322]: Consider reverting these whitespace-only changes to this file.

File: netbox_onboarding/onboarding_extensions/helpers.py [hunk 5]
```diff
@@ -0,0 +1,14 @@
+from enum import Enum
+
+
+class GeneralDeploymentMode(Enum):
```
Comment (glennmatthews) [inline netbox_onboarding/onboarding_extensions/helpers.py:4]: This appears to be unused?

File: netbox_onboarding/tests/test_onboarding_addons_ios.py [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+# Load utilities
+from .utilities import load_test_output
+
+
+class IosExtensionTestCase(TestCase):
+    """Test the IOS NapalmDeviceExtensions."""
+
+    def test_parse_output(self):
+        """Method to test parsing of output through NTC Templates parser"""
+        test_output_dir = "/source/netbox_onboarding/tests/fixtures"
```
Comment (glennmatthews) [inline netbox_onboarding/tests/test_onboarding_addons_ios.py:40]: This shouldn't be a hard-coded absolute path - use `os.path` relative to `__file__` instead - something like `os.path.join(os.path.dirname(__file__), 'fixtures')` I think.

File: README.md (hunk 1 above)
Comment (jvanderaa) [reply README.md:101]: I agree. It made sense in the time of writing.

**Pull request closed without merging**